src/backtrace/mod.rs

```diff
@@ -125,6 +125,37 @@ impl fmt::Debug for Frame {
     }
 }
 
+#[cfg(all(target_env = "sgx", target_vendor = "fortanix", not(feature = "std")))]
+mod sgx_no_std_image_base {
+    use core::sync::atomic::{AtomicU64, Ordering::SeqCst};
+
+    static IMAGE_BASE: AtomicU64 = AtomicU64::new(0);
+
+    /// Set the image base address. This is only available for Fortanix SGX
+    /// target when the `std` feature is not enabled. This can be used in the
+    /// standard library to set the correct base address.
+    pub fn set_image_base(base_addr: u64) {
+        IMAGE_BASE.store(base_addr, SeqCst);
+    }
+
+    pub(crate) fn get_image_base() -> u64 {
+        IMAGE_BASE.load(SeqCst)
+    }
```

Comment:
shouldn't this be usize?

Reply:
it can be changed as such, but [the one in std](https://github.com/rust-lang/rust/blob/1.72.1/library/std/src/sys/sgx/abi/mem.rs#L37) returns `u64`

Reply:
I don't quite understand that either, because the types appropriate for addresses are `usize`, `*const T`, and `*mut T`.

Reply:
The target only supports x86_64